sklearn/compose/_column_transformer.py

```diff
@@ -707,3 +708,15 @@ def make_column_transformer(*transformers, **kwargs):
                              remainder=remainder,
                              sparse_threshold=sparse_threshold,
                              verbose=verbose)
+
+
+def _passthrough_func(X):
+    """
+    Function used in the FunctionTransformer for 'passthrough' columns
+    to ensure the correct shape.
+    """
+    if not getattr(X, 'ndim', 0) == 2:
+        if hasattr(X, 'values'):
```

Comment:
wouldn't simply `X = np.asarray(X)` be better here?

Reply:
it's better but it's unclear to me whether either is good ;)

Reply:
`X = np.asarray(X)` would work on anything that has `__array__` (`NamedArray` for instance), but `values` is specific to `pandas`.

Reply:
also ``values`` does a copy.